sdk/storage/Azure.Storage.Files.DataLake/src/DataLakePathClient.cs

```diff
@@ -1875,31 +1875,40 @@ internal virtual void SetNameFieldsIfNull()
                 try
                 {
                     scope.Start();
-                    ResponseWithHeaders<PathDeleteHeaders> response;
+                    ResponseWithHeaders<PathDeleteHeaders> response = null;
+                    bool paginated = _clientConfiguration.ClientOptions.Version >= DataLakeClientOptions.ServiceVersion.V2023_05_03;
```

Comment:
Waiting to hear back from the service if this is really the min version for this feature.